src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerCorporateValidatorTest.java

```diff
@@ -0,0 +1,384 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import org.apache.commons.lang.StringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.overseasentitiesapi.mocks.AddressMock;
+import uk.gov.companieshouse.overseasentitiesapi.mocks.BeneficialOwnerAllFieldsMock;
+import uk.gov.companieshouse.overseasentitiesapi.model.NatureOfControlType;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.BeneficialOwnerCorporateDto;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationMessages;
+import uk.gov.companieshouse.service.rest.err.Err;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto.BENEFICIAL_OWNERS_CORPORATE_FIELD;
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationUtils.getQualifiedFieldName;
+
+@ExtendWith(MockitoExtension.class)
+class BeneficialOwnerCorporateValidatorTest {
+
+    private static final String CONTEXT = "12345";
+
+    private AddressDtoValidator addressDtoValidator;
+    private BeneficialOwnerCorporateValidator beneficialOwnerCorporateValidator;
+
+    private List<BeneficialOwnerCorporateDto> beneficialOwnerCorporateDtoList;
+
+    @BeforeEach
+    public void init() {
+        addressDtoValidator = new AddressDtoValidator();
+        beneficialOwnerCorporateValidator = new BeneficialOwnerCorporateValidator(addressDtoValidator);
+        beneficialOwnerCorporateDtoList = new ArrayList<>();
+        BeneficialOwnerCorporateDto beneficialOwnerCorporateDto = BeneficialOwnerAllFieldsMock.getBeneficialOwnerCorporateDto();
+        beneficialOwnerCorporateDto.setPrincipalAddress(AddressMock.getAddressDto());
+        beneficialOwnerCorporateDtoList.add(beneficialOwnerCorporateDto);
+    }
+
+    @Test
+    void testNoErrorReportedWhenAllFieldsAreCorrect() {
+        Errors errors = beneficialOwnerCorporateValidator.validate(beneficialOwnerCorporateDtoList, new Errors(), CONTEXT);
+        assertFalse(errors.hasErrors());
+    }
+
+    @Test
+    void testErrorReportedWhenNameFieldIsEmpty() {
+        beneficialOwnerCorporateDtoList.get(0).setName("  ");
+        Errors errors = beneficialOwnerCorporateValidator.validate(beneficialOwnerCorporateDtoList, new Errors(), CONTEXT);
+        String qualifiedFieldName = getQualifiedFieldName(
+                BENEFICIAL_OWNERS_CORPORATE_FIELD,
+                BeneficialOwnerCorporateDto.NAME_FIELD);
+        String validationMessage = ValidationMessages.NOT_EMPTY_ERROR_MESSAGE.replace("%s", qualifiedFieldName);
```

Comment:
Here and below, `String.format(ValidationMessages.NOT_EMPTY_ERROR_MESSAGE, qualifiedFieldName)` is a better (safer) way to do this. See examples in `AddressDtoValidator`, if required.

Reply:
I'd be inclined to leave this and the comment above as it is for now. This looks like it would need to be changed in several places, if you are really sure that it does need to be changed.

Reply:
I think we should change it in the new lines of code you've written as it's the correct way to do it. But older code (not changed in this PR) can remain as is.

Reply:
I have changed this now.